src/transaction.js

```diff
@@ -39,14 +39,27 @@ export class Transaction {
     }
     this._networkPassphrase = networkPassphrase;
 
+    const txe = envelope.value();
+    let sourceAccount;
+    this._envelopeType = envelope.switch();
+    switch (this._envelopeType) {
+      case xdr.EnvelopeType.envelopeTypeTxV0():
+        sourceAccount = txe.tx().sourceAccountEd25519();
+        break;
+      case xdr.EnvelopeType.envelopeTypeTx():
+        sourceAccount = txe
+          .tx()
+          .sourceAccount()
+          .ed25519();
+        break;
+      default:
+        // TODO: Add support for FeeBumpTransaction
+        throw new Error('Invalid transaction.');
```

Comment:
This could be more descriptive, is FeeBump the only one that should fail? Might be nice to have this say something like, 
> Invalid transaction, fee bump transactions are not supported yet. Additional details at `<github issue url>`

Reply:
@vcarl this is temporal, we add fee bump support on this other PR  #321